src/types/types.ts

```diff
@@ -1,26 +1,27 @@
 import { Request } from "express";
 import { EntityManager } from "@mikro-orm/core";
+import { ObjectId } from "mongoose";
 
 export interface User {
   id: string;
   username: string;
   email: string;
 }
 
-export interface CartItem {
-  product: Product;
+export interface CartItemType {
+  product: ProductType;
   count: number;
 }
 
-export interface Cart {
-  id: string; // uuid
-  userId: string;
+export interface CartType {
+  _id: ObjectId;
```

Comment:
it's not required to declare the _id in the interface, as it'll declared by mongo